vispy/app/backends/_tk.py

```diff
@@ -0,0 +1,701 @@
+KEY_STATE_MAP = {
+    0x0001: keys.SHIFT,
+    # 0x0002: CAPSLOCK,
+    0x0004: keys.CONTROL,
+    # 0x0008: keys.ALT,  # LEFT_ALT: Seems always pressed?
+    # 0x0010: NUMLOCK,
+    # 0x0020: SCROLLLOCK,
+    0x0080: keys.ALT,
+    # 0x0100: ?,  # Mouse button 1.
+    # 0x0200: ?,  # Mouse button 2.
+    # 0x0400: ?,  # Mouse button 3.
+    0x20000: keys.ALT  # LEFT_ALT ?
+}
+
+# e.num -> vispy
+MOUSE_BUTTON_MAP = {
+    1: 1,  # Mouse Left   == 1 -> Mouse Left
+    2: 3,  # Mouse Middle == 2 -> Mouse Middle
+    3: 2,  # Mouse Right  == 3 -> Mouse Right
+
+    # TODO: If other mouse buttons are needed
+    # and they differ from the Tkinter numbering, add them here.
+    # e.g. BACK/FORWARD buttons or other custom mouse buttons
+}
+
+# -------------------------------------------------------------- capability ---
+
+# These are all booleans. Note that they mirror many of the kwargs to
+# the initialization of the Canvas class.
+capability = dict(
+    # if True they mean:
+    title=True,           # can set title on the fly
+    size=True,            # can set size on the fly
+    position=True,        # can set position on the fly
+    show=True,            # can show/hide window
+    vsync=False,          # can set window to sync to blank
+    resizable=True,       # can toggle resizability (e.g., no user resizing)
+    decorate=True,        # can toggle decorations
+    fullscreen=True,      # fullscreen window support
+    context=False,        # can share contexts between windows
+    multi_window=True,    # can use multiple windows at once
+    scroll=True,          # scroll-wheel events are supported
+    parent=True,          # can pass native widget backend parent
+    always_on_top=True,   # can be made always-on-top
+)
+
+
+# ------------------------------------------------------- set_configuration ---
+def _set_config(c):
+    """Set gl configuration for template.
+    Currently not used for Tkinter backend.
+    """
+    return []
+
+
+# ------------------------------------------------------------- application ---
+
+_tk_inst = None         # Reference to tk.Tk instance
+_tk_inst_owned = False  # Whether we created the Tk instance or not
+_tk_toplevels = []      # References to created CanvasBackend Toplevels
+
+
+def _new_toplevel(self, *args, **kwargs):
+    """
+    Create and return a new withdrawn Toplevel.
+    Create a tk.Toplevel with the given args and kwargs,
+    minimize it and add it to the global list before returning.
+
+    :return: Return the created tk.Toplevel
+    """
+    global _tk_inst, _tk_toplevels
+    tl = tk.Toplevel(_tk_inst, *args, **kwargs)
+    tl.withdraw()
+    _tk_toplevels.append(self)
+    return tl
+
+
+def _del_toplevel(fr=None):
+    """
+    Destroy the given Toplevel, and if it was the last one,
+    also destroy the global Tk instance if we created it.
+
+    :param fr: The CanvasBackend to destroy, defaults to None
+    """
+    global _tk_inst, _tk_inst_owned, _tk_toplevels
```

Comment:
```suggestion
    global _tk_inst
```

Reply:
Globals are moved to their own class `_TkInstanceManager`.